lib/fabrik/VirtualHostAgent.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Agent = require('./Agent');
+
+class VirtualHostAgent extends Agent {
+
+  constructor(settings) {
+    super({
+      json: true
+    });
+    this.settings = settings;
+  }
+
+  provision(ips, guid) {
+    const body = {
+      parameters: {}
+    };
+    return this
+      .getHost(ips, 'multi-tenancy')
+      .then(ip => this.post(ip, `tenants/${guid}/lifecycle/provision`, body, 200));
+  }
+
+  deprovision(ips, guid) {
+    const body = {
+      parameters: {}
+    };
+    return this
+      .getHost(ips, 'multi-tenancy')
+      .then(ip => this.post(ip, `tenants/${guid}/lifecycle/deprovision`, body, 200));
+  }
+
+  createCredentials(ips, guid, parameters) {
+    const body = {
+      parameters: parameters
+    };
+    return this
+      .getHost(ips, 'credentials')
+      .then(ip => this.post(ip, `tenants/${guid}/credentials/create`, body, 200));
+  }
+
+  deleteCredentials(ips, guid, credentials) {
+    const body = {
+      credentials: credentials
+    };
+    return this
+      .getHost(ips, 'credentials')
+      .then(ip => this.post(ip, `tenants/${guid}/credentials/delete`, body, 200));
```

Comment:
wrong HTTP semantics/ bad practice. Use HTTP Delete on resource `tenants/${guid}/credentials/`
suffix `/delete` can be omitted, its redundant.

Reply:
For delete credentials, I tried keeping URL format similar to what is listed here, https://cloudfoundry-incubator.github.io/service-fabrik-broker/api/agent_v1.1.html. Same is used https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/lib/fabrik/Agent.js#L138
Should I change the format?

Reply:
same as [above](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/111/files#r157019650)